packages/block-library/src/post-excerpt/index.php

```diff
@@ -24,7 +24,14 @@ function render_block_core_post_excerpt( $attributes, $content, $block ) {
 		return '';
 	}
 
-	$more_text           = ! empty( $attributes['moreText'] ) ? '<a class="wp-block-post-excerpt__more-link" href="' . esc_url( get_the_permalink( $block->context['postId'] ) ) . '">' . wp_kses_post( $attributes['moreText'] ) . '</a>' : '';
+	$post_ID            = $block->context['postId'];
+	$post_title         = get_the_title( $post_ID );
+	$screen_reader_text = sprintf(
+		/* translators: %s is either the post title or post ID to describe the link for screen readers. */
+		__( ': %s' ),
+		'' !== $post_title ? $post_title : __( 'untitled post ' ) . $post_ID
+	);
+	$more_text           = ! empty( $attributes['moreText'] ) ? '<a class="wp-block-post-excerpt__more-link" href="' . esc_url( get_the_permalink( $post_ID ) ) . '">' . wp_kses_post( $attributes['moreText'] ) . '<span class="screen-reader-text">' . $screen_reader_text . '</span></a>' : '';
```

Comment:
I am pretty sure we should be filtering the output here. Probably need to fix this on the other PR.

```suggestion
	$more_text           = ! empty( $attributes['moreText'] ) ? '<a class="wp-block-post-excerpt__more-link" href="' . esc_url( get_the_permalink( $post_ID ) ) . '">' . wp_kses_post( $attributes['moreText'] ) . '<span class="screen-reader-text">' . wp_kses_post( $screen_reader_text ) . '</span></a>' : '';
```

Reply:
Thanks for the review.

Since the post title can contain html, do we want `wp_kses_post`, or `esc_html` instead? This way, we're safely displaying the title as well as removing any tags that may exist.

Reply:
@jffng I think my suggestion is fine for this as it is used for `$attributes['moreText']`. Although this may be an area one of the Core Committers should have a look at.

Reply:
@carolinan do you have any thoughts on the escaping / filtering that may be needed here?

Reply:
`esc_html` would potentially leave encoded entities in the string, which could potentially be a lousy reading experience. `wp_kses_post` may not be aggressive enough, as it would leave tags that aren't allowed inside a `span`, e.g., leaving a `div`.  Maybe `wp_kses` with a passed array of only inline tags?